### PR TITLE
update slide_puzzle

### PR DESCRIPTION
**Pull request by johnpryan**
add missing files from https://github.com/flutter/samples/pull/419

### CODE REVIEW

File: web/slide_puzzle/lib/src/core/puzzle_proxy.dart [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import 'dart:math' show Point;
```
Comment (RedBrogdon) [inline web/slide_puzzle/lib/src/core/puzzle_proxy.dart:1]: Copyright header on this and the other new files?

Comment (johnpryan) [reply web/slide_puzzle/lib/src/core/puzzle_proxy.dart:1]: Hmm... Do we need copyright headers if we aren't the original author?

Comment (RedBrogdon) [reply web/slide_puzzle/lib/src/core/puzzle_proxy.dart:1]: What was the original source for the slide puzzle? Was this something @yjbanov or @kevmoo did?

**Comment by kevmoo** [comment]
It's my project. Look under might github user to find it

File: web/slide_puzzle/lib/src/core/puzzle_proxy.dart (hunk 1 above)
Comment (johnpryan) [reply web/slide_puzzle/lib/src/core/puzzle_proxy.dart:1]: https://github.com/kevmoo/slide_puzzle

Comment (RedBrogdon) [reply web/slide_puzzle/lib/src/core/puzzle_proxy.dart:1]: In that case, for consistency I would go ahead and add headers for consistency with The Flutter Team or The Dart Project listed as the author.

**Comment by kevmoo** [comment]
The code was paid for by Dart/Flutter :-)

**Comment by johnpryan** [comment]
@kevmoo I'll send you a PR to add the headers and copy those updates into this PR

**Comment by johnpryan** [comment]
I added 3d8687e to this change, which applies https://github.com/kevmoo/slide_puzzle/pull/6.

**Pull request merged**